File: app/src/main/java/org/bspb/smartbirds/pro/ui/SplashScreenActivity.java
package org.bspb.smartbirds.pro.ui;

import android.content.res.Configuration;
import android.os.Bundle;
import android.view.View;
import android.view.Window;

import androidx.annotation.Nullable;
import androidx.core.app.TaskStackBuilder;

import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Fullscreen;
import org.androidannotations.annotations.ViewById;
import org.androidannotations.annotations.WindowFeature;
import org.androidannotations.annotations.sharedpreferences.Pref;
import org.bspb.smartbirds.pro.R;
import org.bspb.smartbirds.pro.prefs.SmartBirdsPrefs_;
import org.bspb.smartbirds.pro.service.DataService_;

@WindowFeature({Window.FEATURE_NO_TITLE})
@Fullscreen
@EActivity(R.layout.activity_splash_screen)
public class SplashScreenActivity extends BaseActivity implements Runnable {

    @ViewById(android.R.id.content)
    View content;

    @Pref
    SmartBirdsPrefs_ prefs;

    // Duration in milliseconds
    private static final int SPLASH_DURATION = 3000;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onResume() {
        super.onResume();
        content.postDelayed(this, SPLASH_DURATION);
    }

    @Override
    protected void onPause() {
        content.removeCallbacks(this);
        super.onPause();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        setContentView(R.layout.activity_splash_screen);
    }

    @Override
    public void run() {
        if (prefs.runningMonitoring().get()) {
            DataService_.intent(this).start();
            TaskStackBuilder.create(SplashScreenActivity.this)
                    .addNextIntentWithParentStack(MonitoringActivity_.intent(SplashScreenActivity.this).get())
                    .startActivities();
        } else {
            MainActivity_.intent(SplashScreenActivity.this).start();
        }

        finish();
    }
}

File: app/src/main/kotlin/org/bspb/smartbirds/pro/ui/SplashScreenActivity.kt
package org.bspb.smartbirds.pro.ui

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import androidx.core.app.TaskStackBuilder
import org.bspb.smartbirds.pro.R
import org.bspb.smartbirds.pro.prefs.SmartBirdsPrefs
import org.bspb.smartbirds.pro.service.DataService

open class SplashScreenActivity : BaseActivity(), Runnable {

    protected lateinit var content: View

    protected lateinit var prefs: SmartBirdsPrefs

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportRequestWindowFeature(1)

        setContentView(R.layout.activity_splash_screen)

        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )

        prefs = SmartBirdsPrefs(this)
        content = findViewById(android.R.id.content)
    }

    override fun onResume() {
        super.onResume()
        content.postDelayed(this, SPLASH_DURATION.toLong())
    }

    override fun onPause() {
        content.removeCallbacks(this)
        super.onPause()
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        setContentView(R.layout.activity_splash_screen)
    }

    override fun run() {
        if (prefs.getRunningMonitoring()) {
            DataService.intent(this).start()
            TaskStackBuilder.create(this@SplashScreenActivity)
                .addNextIntentWithParentStack(
                    MonitoringActivity.newIntent(this@SplashScreenActivity)
                )
                .startActivities()
        } else {
            startActivity(Intent(this@SplashScreenActivity, MainActivity::class.java))
        }
        finish()
    }

    companion object {
        // Duration in milliseconds
        private const val SPLASH_DURATION = 3000
    }
}